eZ/Publish/Core/Repository/BookmarkService.php

```diff
@@ -97,16 +100,22 @@ public function loadBookmarks(int $offset = 0, int $limit = 25): BookmarkList
     {
         $currentUserId = $this->getCurrentUserId();
 
-        $list = new BookmarkList();
-        $list->totalCount = $this->bookmarkHandler->countUserBookmarks($currentUserId);
-        if ($list->totalCount > 0) {
-            $bookmarks = $this->bookmarkHandler->loadUserBookmarks($currentUserId, $offset, $limit);
-
-            $list->items = array_map(function (Bookmark $bookmark) {
-                return $this->repository->getLocationService()->loadLocation($bookmark->locationId);
-            }, $bookmarks);
+        $filter = new Filter();
+        try {
+            $filter
+                ->withCriterion(new Criterion\IsBookmarked($currentUserId))
+                ->withSortClause(new SortClause\BookmarkId(Query::SORT_DESC))
+                ->sliceBy($limit, $offset);
+
+            $result = $this->repository->getlocationService()->find($filter, []);
+        } catch (\eZ\Publish\API\Repository\Exceptions\BadStateException $e) {
```

Comment:
This class should be imported.

```suggestion
        } catch (BadStateException $e) {
```

Reply:
Fixed